addons/mail/models/mail_thread.py

```diff
@@ -221,10 +221,10 @@ def _compute_message_has_error(self):
         if self.ids:
             self._cr.execute(""" SELECT msg.res_id, COUNT(msg.res_id) FROM mail_message msg
                                  RIGHT JOIN mail_notification rel
-                                 ON rel.mail_message_id = msg.id AND rel.notification_status in ('exception','bounce')
+                                 ON rel.mail_message_id = msg.id AND rel.notification_status in ('exception','bounce') AND rel.author_id = %s
```

Comment:
* Why isn't `rel.author_id = %s` in where clause?
* I know you don't changed the previous line with `RIGHT JOIN` but how a rel can have a NULL msg with a given author and message?

Reply:
About having it in `ON` instead of `WHERE`, isn't it equivalent?

Condition on `notification_status` was in `ON` on the existing query so I felt the condition on author should be next to it, as they go together conceptually in what we are trying to do here.
If they are better in the `WHERE` part, shouldn't they both be moved then?

About the `RIGHT JOIN` and the `NULL` I have no idea what you are asking, isn't it a comment meant for another line of code?

Reply:
for the right join, i'm talking about the line just above your diff, the one from the ON clause you're modifying.

Reply:
Are you asking that I'm making it `INNER JOIN` explicitly instead because the right part will have no record that would not be on the inner part?